crates/networking/src/stream.rs

```diff
@@ -0,0 +1,93 @@
+use std::io::{BufRead, BufReader, Read};
+use std::marker::PhantomData;
+
+use oreo_errors::OreoError;
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use ureq::Response;
+
+use crate::rpc_abi::RpcResponseStream;
+
+pub struct StreamReader<T> {
+    reader: BufReader<Box<dyn Read>>,
```

Comment:
If you change `StreamReader<T>` into `StreamReader<T, R>`, then you can get rid of the box and dynamic dispatch:

```rust
pub struct StreamReader<T, R> {
    reader: BufReader<R>,
    _marker: PhantomData<T>,
}
```

Reply:
Ninja